sematic/db/models/external_resource_record.py

```diff
@@ -0,0 +1,168 @@
+# Standard Library
+from typing import Any, Dict, Tuple, Type, Union
+
+# Third-party
+from sqlalchemy import Column, types
+from sqlalchemy.orm import validates
+
+# Sematic
+from sematic.db.models.base import Base
+from sematic.db.models.json_encodable_mixin import JSONEncodableMixin
+from sematic.external_resource import ExternalResource, ResourceState
+from sematic.types.serialization import (
+    type_from_json_encodable,
+    type_to_json_encodable,
+    value_from_json_encodable,
+    value_to_json_encodable,
+)
+
+TypeSerialization = Dict[str, Any]
+
+
+class ExternalResourceRecord(Base, JSONEncodableMixin):
+    """A DB record for an ExternalResource and its history.
+
+    Attributes
+    ----------
+    id:
+        The unique id of the external resource
+    resource_state:
+        The current state of the resource
+    locally_allocated:
+        Whether or not the resource was locally allocated
```

Comment:
Can you explain what that means and why it's useful?

Reply:
discussed offline. I'm changing to `managed_by` which can be one of `LOCAL`, `REMOTE`, or `UNKNOWN`. It is now a part of the `ResourceStatus`.